lib/elasticity/segmented_document.rb

```diff
@@ -0,0 +1,33 @@
+module Elasticity
+  class SegmentedDocument < BaseDocument
+    # Creates a new segment which behaves almost the same as a Document class dynamically
+    # configured to access a segmented index.
+    #
+    # It creates a new class in runtime that inherits from your defined class, allowing
+    # methods defined in your class to be callable from the dynamic class.
+    def self.segment(segment_name)
+      qn = segment_name.camelize
+
+      klass = Class.new(self) do
+        class_attribute :mapper, :segment_name
+        IndexMapper.set_delegates(singleton_class, :mapper)
```

Comment:
Feels unusual to use IndexMapper to set up delegation here instead of just including a module that delegates the methods. Is there a benefit to doing it this way?